dds/src/implementation/rtps/types.rs

```diff
@@ -91,15 +91,15 @@ impl TryFrom<BuiltInTopicKey> for Guid {
 pub struct GuidPrefix([u8; 12]);
 pub const GUIDPREFIX_UNKNOWN: GuidPrefix = GuidPrefix([0; 12]);
 
-impl From<GuidPrefix> for [u8; 12] {
-    fn from(value: GuidPrefix) -> Self {
-        value.0
+impl GuidPrefix {

```

Comment:
We could potentially save ourselves a little bit of code by using the derive_more crate to derive all these implementations.